LookingMenu/Services/SQLiteService.swift

```diff
@@ -0,0 +1,146 @@
+            sqlite3_bind_text(statement, 1, (name as NSString).utf8String, -1, nil)
+            sqlite3_bind_double(statement, 2, Double(calor))
+            let data = try! JSONEncoder().encode(recipeSessions)
+            let listString = String(data: data, encoding: .utf8)
+            
+            sqlite3_bind_text(statement, 3, (listString! as NSString).utf8String, -1, nil)
+            sqlite3_step(statement)
+        } else {
+            print("Query is not as per requirement")
+        }
+    }
+    
+    func readTableDiet() -> [Diet] {
+        var result = [Diet]()
+        let query = ConstantSqlite.queryReadDataDiet
+        var statement : OpaquePointer? = nil
+        if sqlite3_prepare_v2(db, query, -1, &statement, nil) == SQLITE_OK{
+            while sqlite3_step(statement) == SQLITE_ROW {
+                let id = Int(sqlite3_column_int(statement, 0))
+                let name = String(describing: String(cString: sqlite3_column_text(statement, 1)))
+                let calor = Double(sqlite3_column_double(statement, 2))
+                let list = String(describing: String(cString: sqlite3_column_text(statement, 3)))
+                guard let data = try? JSONDecoder().decode([RecipeSession].self,
+                                                           from: list.data(using: .utf8)!)
+                else { return result }
+                let element = Diet(id: id,
+                                   name: name,
+                                   calor: calor,
+                                   recipeSessions: data)
+                result.append(element)
+            }
+        }
+        return result
+    }
+    
+    func deleteDiet(idDiet : Int) {
+        let query = "DELETE FROM Diet WHERE id=?;"
+        var statement : OpaquePointer? = nil
+        if sqlite3_prepare_v2(db, query, -1, &statement, nil) == SQLITE_OK{
+            sqlite3_bind_int(statement, 1, Int32(idDiet))
+            sqlite3_step(statement)
+        }
+        sqlite3_finalize(statement)
+    }
+    
+    func checkRecipeFavourite(idDiet : Int) -> Bool{
+        var result = false
+        let query = ConstantSqlite.queryCheckDataDietFavourite
+        var statement: OpaquePointer? = nil
+        if sqlite3_prepare_v2(db, query, -1, &statement, nil) ==
+            SQLITE_OK {
+            sqlite3_bind_int(statement, 1, Int32(idDiet))
+            if sqlite3_step(statement) == SQLITE_ROW {
+                result = true
+            }
+        }
+        sqlite3_finalize(statement)
+        return result
+    }
+    
+    func insertRecipeFavourite(id: Int, title : String, readyInMinutes : Int, image: String) {
+        let query = ConstantSqlite.queryInsertDataDietFavourite
+        var statement: OpaquePointer? = nil
+        if sqlite3_prepare_v2(db, query, -1, &statement, nil) == SQLITE_OK {
+            sqlite3_bind_int(statement, 1, Int32(id))
+            sqlite3_bind_text(statement, 2, (title as NSString).utf8String, -1, nil)
```

Comment:
có cần thiết phải dùng đến `NSString` không em?

Reply:
Dạ có a ơi . Do ở đó bind text nên phải dùng cái NSString

Reply:
dùng cách của em ok. Có thể tham khảo thêm cái này nhé:
https://stackoverflow.com/questions/37087325/how-to-convert-string-to-unicodeutf-8-string-in-swift